File: mycuration/src/test/java/com/phicdy/mycuration/util/test/UrlUtilTest.java
package com.phicdy.mycuration.util.test;

import com.phicdy.mycuration.util.UrlUtil;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.junit.Assert.*;

@RunWith(JUnit4.class)
public class UrlUtilTest {

	public UrlUtilTest() {
		super();
	}

	@Test
	public void testRemoveUrlParameter() {
		String removedUrl = UrlUtil.INSTANCE.removeUrlParameter("http://harofree.blog.fc2.com/?ps");
		assertEquals("http://harofree.blog.fc2.com/", removedUrl);
	}

	@Test
	public void testHasParameterUrl() {
		assertEquals(true, UrlUtil.INSTANCE.hasParameterUrl("http://www.xxx.com/?aaa"));
		assertEquals(true, UrlUtil.INSTANCE.hasParameterUrl("https://www.xxx.com/?aaa"));
		assertEquals(true, UrlUtil.INSTANCE.hasParameterUrl("http://www.xxx.com/aaa/?bbb"));
		assertEquals(false, UrlUtil.INSTANCE.hasParameterUrl("http://www.xxx.com/aaa"));
		assertEquals(false, UrlUtil.INSTANCE.hasParameterUrl("http://www.xxx.com?/aaa"));
	}
}

File: util/src/test/java/com/phicdy/mycuration/util/UrlUtilTest.kt
package com.phicdy.mycuration.util

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class UrlUtilTest {

    @Test
    fun testRemoveUrlParameter() {
        val removedUrl = UrlUtil.removeUrlParameter("http://harofree.blog.fc2.com/?ps")
        assertThat(removedUrl).isEqualTo("http://harofree.blog.fc2.com/")
    }

    @Test
    fun testHasParameterUrl() {
        assertThat(UrlUtil.hasParameterUrl("http://www.xxx.com/?aaa")).isTrue()
        assertThat(UrlUtil.hasParameterUrl("https://www.xxx.com/?aaa")).isTrue()
        assertThat(UrlUtil.hasParameterUrl("http://www.xxx.com/aaa/?bbb")).isTrue()
        assertThat(UrlUtil.hasParameterUrl("http://www.xxx.com/aaa")).isFalse()
        assertThat(UrlUtil.hasParameterUrl("http://www.xxx.com?/aaa")).isFalse()
    }
}
